### ISSUE TITLE
Link to download should be checked/updated on a regular basis

### ISSUE THREAD

**Issue by kaimeinke** [oceanprotocol/market#275]
The information and the download is just updated once when a dataset is beeing created. It fetched the file size, format and availability.

In my opinion this should take place at least once every 24h or so in order to:

- show the correct file size
- alert users if the resource is broken/missing
- this would protect liquidity providers against bad actors that remove their data from their hosting platform
- this would protect liquidity providers against technical problems that just might happen
- this could inform data providers if there are any problems with their offered date to communicate fixes

Example: This file has been update several times and it now much bigger as the dataset grows. This outdated size information irritates users and buyers and might negatively affect the asset.

![image](https://user-images.githubusercontent.com/59994024/100346617-6b0d1300-2fe4-11eb-9c3e-aa82b5390cd8.png)

**Comment by alexcos20** [comment]
in order to update this information, metadata needs to be updated. But only the publisher can do this, because he is the only one who can trigger an update on the chain

**Comment by kaimeinke** [comment]
Okay, thanks for the quick reply. Could the provider then manually trigger such a ping in theory?

**Comment by alexcos20** [comment]
yes, but the provider is not able to update the metadata.  So it has to be something like:
- Publisher triggers provider update
- Provider returns new size to the publisher
- Publisher is updating the metadata by generating a new transaction on-chain

**Comment by kaimeinke** [comment]
I see. I confused provider with publisher as well. I think the publisher / creator of the pool could be the one to trigger such an update, even if it might include gas fees. This could be a part of regular pool maintenance. The reward would be a signal that the pool is fresh / updated / valid.

**Comment by alexcos20** [comment]
agreed. we need a method in the libs first for this and only after that we can have in the mp.  I will take it further
